### PR TITLE
Apply example_ynh

### PR DESCRIPTION
**Pull request by yalh76**
Apply example_ynh

### CODE REVIEW

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
!testme

File: conf/share-smb.conf [hunk 1]
```diff
@@ -4,7 +4,7 @@
    path = /home/yunohost.app/samba/share
    guest ok = no
    browsable = yes
-   valid users = @samba.share
+   valid users = @sambashare
```
Comment (zamentur) [inline conf/share-smb.conf:7]:
```suggestion
   valid users = @samba.share
```

We want the permission group Samba (share) to be added to the user authorized. So the dot is important.

File: scripts/install [hunk 2]
```diff
@@ -79,7 +71,7 @@ mkdir -p $datadir/share
 chmod 750 "$datadir/share"
 chmod -R o-rwx "$datadir/share"
 chown -R root:root "$datadir/share"
-setfacl -R -m g:samba.share:rwx,d:g:samba.share:rwx $datadir/share
+setfacl -R -m g:sambashare:rwx,d:g:sambashare:rwx $datadir/share
```
Comment (zamentur) [inline scripts/install:74]:
```suggestion
setfacl -R -m g:samba.share:rwx,d:g:samba.share:rwx $datadir/share
```

File: scripts/restore [hunk 3]
```diff
@@ -69,12 +75,20 @@ ynh_script_progression --message="Reinstalling dependencies..." --weight=1
...
+
+
+#=================================================
 # INTEGRATE SERVICE IN YUNOHOST
-#================================================
+#=================================================
 ynh_script_progression --message="Integrating service in YunoHost..." --weight=1
 
-yunohost service add smbd --description="Samba service" --log="/var/log/smbd/smbd.log"
+yunohost service add smbd --description="Samba service" --log="/var/log/smbd/smbd.log" --needs_exposed_ports 445
```
Comment (zamentur) [inline scripts/restore:91]: This params will be used by remote diagnosis to remotely test if the port is open and the diagnosis will say it should be open. BUT we don't want to open that port on the internet.

Comment (zamentur) [reply scripts/restore:91]:
```suggestion
# Don't add --needs_exposed_ports 445 , we don't want a remote diagnosis here
yunohost service add smbd --description="Samba service" --log="/var/log/smbd/smbd.log"
```

File: scripts/upgrade [hunk 4]
```diff
@@ -45,19 +47,26 @@ ynh_script_progression --message="Stopping a systemd service..." --weight=4
...
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
 # INTEGRATE SERVICE IN YUNOHOST
-#================================================
+#=================================================
 ynh_script_progression --message="Integrating service in YunoHost..." --weight=1
 
-yunohost service add smbd --description="Samba service" --log="/var/log/smbd/smbd.log"
+yunohost service add smbd --description="Samba service" --log="/var/log/smbd/smbd.log" --needs_exposed_ports 445
```
Comment (zamentur) [inline scripts/upgrade:69]:
```suggestion
# Don't add --needs_exposed_ports 445 , we don't want a remote diagnosis here
yunohost service add smbd --description="Samba service" --log="/var/log/smbd/smbd.log"
```

File: scripts/install [hunk 5]
```diff
@@ -61,11 +57,7 @@ ynh_script_progression --message="Installing dependencies..." --weight=10
 ynh_install_app_dependencies $pkg_dependencies
 
 #=================================================
-# SETUP SSOWAT
-#=================================================
-
-ynh_permission_create --permission="share" --allowed=all_users
-
+# SPECIFIC SETUP
```
Comment (zamentur) [inline scripts/install:60]:
```suggestion
# SPECIFIC SETUP
#=================================================
# SETUP SSOWAT 
#=================================================
ynh_script_progression --message="Configuring permissions..."

# We define the permission in advance to be able to use the group
# in unix permission
ynh_permission_create --permission="share" --allowed=all_users
```

**Review by zamentur — CHANGES_REQUESTED** [review]

**Comment by zamentur** [comment]
!testme

File: conf/share-smb.conf (hunk 1 above)
Comment (yalh76) [reply conf/share-smb.conf]: Fine, but when doing some tests, having `@samba.share` failed ...

**Pull request merged**